### PR TITLE
A new example to demonstrate custom tiled processing/writing

### PR DESCRIPTION
**Pull request by dzenanz**
This example has C++ and Python code.

### CODE REVIEW

**Comment by dzenanz** [comment]
Windows build and test failure log:
```text
2020-08-19T14:06:43.1704466Z [876/1133] Linking CXX executable bin\ApplyAtanImageFilter.exe
2020-08-19T14:06:43.2348113Z ninja: build stopped: interrupted by user.
2020-08-19T14:06:43.3637624Z Traceback (most recent call last):
2020-08-19T14:06:43.3640354Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\runpy.py", line 193, in _run_module_as_main
2020-08-19T14:06:43.3640821Z     "__main__", mod_spec)
2020-08-19T14:06:43.3641176Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\runpy.py", line 85, in _run_code
2020-08-19T14:06:43.3641480Z     exec(code, run_globals)
2020-08-19T14:06:43.3641811Z   File "C:\hostedtoolcache\windows\Python\3.7.8\x64\Scripts\ninja.exe\__main__.py", line 7, in <module>
2020-08-19T14:06:43.3642471Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\site-packages\ninja\__init__.py", line 50, in ninja
2020-08-19T14:06:43.3642829Z     raise SystemExit(_program('ninja', sys.argv[1:]))
2020-08-19T14:06:43.3643174Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\site-packages\ninja\__init__.py", line 46, in _program
2020-08-19T14:06:43.3643483Z     return subprocess.call([os.path.join(BIN_DIR, name)] + args)
2020-08-19T14:06:43.3643817Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\subprocess.py", line 341, in call
2020-08-19T14:06:43.3644364Z     return p.wait(timeout=timeout)
2020-08-19T14:06:43.3644691Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\subprocess.py", line 1019, in wait
2020-08-19T14:06:43.3644983Z     return self._wait(timeout=timeout)
2020-08-19T14:06:43.3645297Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\subprocess.py", line 1259, in _wait
2020-08-19T14:06:43.3645576Z     timeout_millis)
2020-08-19T14:06:43.3646417Z KeyboardInterrupt
2020-08-19T14:06:43.3756295Z Terminate batch job (Y/N)? 
2020-08-19T14:06:43.3763748Z ^C
2020-08-19T14:06:43.4476129Z ##[error]The operation was canceled.
```
On MacOS:
```text
2020-08-19T14:06:26.6786090Z 99% tests passed, 1 tests failed out of 588
2020-08-19T14:06:26.6787000Z 
2020-08-19T14:06:26.6787330Z Label Time Summary:
2020-08-19T14:06:26.6787670Z RUNS_LONG    =  21.80 sec*proc (3 tests)
2020-08-19T14:06:26.6787840Z 
2020-08-19T14:06:26.6788140Z Total Test time (real) = 239.68 sec
2020-08-19T14:06:26.6788320Z 
2020-08-19T14:06:26.6788500Z The following tests FAILED:
2020-08-19T14:06:26.6789450Z 	166 - MutualInformationAffineTest (Failed)
```
I think this is ready for merging.

**Comment by thewtex** [comment]
> 2020-08-19T14:06:43.1704466Z [876/1133] Linking CXX executable bin\ApplyAtanImageFilter.exe
> 2020-08-19T14:06:43.2348113Z ninja: build stopped: interrupted by user.
> 2020-08-19T14:06:43.3637624Z Traceback (most recent call last):
> 2020-08-19T14:06:43.3640354Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\runpy.py", line 193, in _run_module_as_main
> 2020-08-19T14:06:43.3640821Z     "__main__", mod_spec)
> 2020-08-19T14:06:43.3641176Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\runpy.py", line 85, in _run_code
> 2020-08-19T14:06:43.3641480Z     exec(code, run_globals)
> 2020-08-19T14:06:43.3641811Z   File "C:\hostedtoolcache\windows\Python\3.7.8\x64\Scripts\ninja.exe\__main__.py", line 7, in <module>
> 2020-08-19T14:06:43.3642471Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\site-packages\ninja\__init__.py", line 50, in ninja
> 2020-08-19T14:06:43.3642829Z     raise SystemExit(_program('ninja', sys.argv[1:]))
> 2020-08-19T14:06:43.3643174Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\site-packages\ninja\__init__.py", line 46, in _program
> 2020-08-19T14:06:43.3643483Z     return subprocess.call([os.path.join(BIN_DIR, name)] + args)
> 2020-08-19T14:06:43.3643817Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\subprocess.py", line 341, in call
> 2020-08-19T14:06:43.3644364Z     return p.wait(timeout=timeout)
> 2020-08-19T14:06:43.3644691Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\subprocess.py", line 1019, in wait
> 2020-08-19T14:06:43.3644983Z     return self._wait(timeout=timeout)
> 2020-08-19T14:06:43.3645297Z   File "c:\hostedtoolcache\windows\python\3.7.8\x64\lib\subprocess.py", line 1259, in _wait
> 2020-08-19T14:06:43.3645576Z     timeout_millis)
> 2020-08-19T14:06:43.3646417Z KeyboardInterrupt
> 2020-08-19T14:06:43.3756295Z Terminate batch job (Y/N)? 
> 2020-08-19T14:06:43.3763748Z ^C
> 2020-08-19T14:06:43.4476129Z ##[error]The operation was canceled.

Perhaps a [`timeout`](https://docs.python.org/3/library/subprocess.html) argument is required for the `ninja.exe` `subprocess` invocation? @jcfr what do you think?

**Comment by jcfr** [comment]
Which problem are we trying to solve ?

**Comment by thewtex** [comment]
> Which problem are we trying to solve ?

It appears that the `ninja` invocation is timing out on long builds on Windows.

File: src/IO/ImageBase/ProcessImageChunks/Code.cxx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        // convert region into IO region
+        itk::ImageIORegion ioRegion(Dimension);
+        ioRegion.SetIndex(0, start[0]);
+        ioRegion.SetIndex(1, start[1]);
+        ioRegion.SetIndex(2, start[2]);
+        ioRegion.SetSize(0, region.GetSize()[0]);
+        ioRegion.SetSize(1, region.GetSize()[1]);
+        ioRegion.SetSize(2, region.GetSize()[2]);
+        // tell the writer this is only a chunk of a larger image
```
Comment (blowekamp) [inline src/IO/ImageBase/ProcessImageChunks/Code.cxx:112]: The ImageRegionSplitterMultidimensional provides vary similar functionality to this manual splitting done above. I see the comment reference "control". Perhaps at build in ITK functionality should at least be referenced even if it's not used.

Comment (dzenanz) [reply src/IO/ImageBase/ProcessImageChunks/Code.cxx:112]: OK, I guess I will also add a more standard ITK streaming pipeline, which has less manual code but would probably be less clear to newcomers.

**Comment by dzenanz** [comment]
@curtislisle take a look 😄

File: src/IO/ImageBase/ProcessImageChunks/Code.cxx (hunk 1 above)
Comment (dzenanz) [reply src/IO/ImageBase/ProcessImageChunks/Code.cxx:116]: I realized that https://github.com/InsightSoftwareConsortium/ITKExamples/blob/master/src/Core/Common/StreamAPipeline/Code.cxx already demonstrates a regular pipeline. Also, this example is meant more for use from Python as part of ML approach. I updated the comments accordingly.

**Pull request merged**